### PR TITLE
[2.7] Static Code Analysis for Components and Bundles

### PR DESCRIPTION
**Pull request by kalessil**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Static Code Analysis with Php Inspections (EA Extended):
    - all rules accepted into 2.3

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/containers/container9.php [hunk 1]
```diff
@@ -50,7 +50,7 @@
     ->addMethodCall('setBar', array(new Reference('foo2', ContainerInterface::NULL_ON_INVALID_REFERENCE)))
     ->addMethodCall('setBar', array(new Reference('foo3', ContainerInterface::IGNORE_ON_INVALID_REFERENCE)))
     ->addMethodCall('setBar', array(new Reference('foobaz', ContainerInterface::IGNORE_ON_INVALID_REFERENCE)))
-    ->addMethodCall('setBar', array(new Expression('service("foo").foo() ~ (container.hasparameter("foo") ? parameter("foo") : "default")')))
+    ->addMethodCall('setBar', array(new Expression("service('foo').foo() ~ (container.hasParameter('foo') ? parameter('foo') : 'default')")))
```
Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/containers/container9.php]: What is the reason for this change?

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services6.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ services:
         calls:
             - [ setBar, [] ]
             - [ setBar ]
-            - [ setBar, ['@=service("foo").foo() ~ (container.hasparameter("foo") ? parameter("foo") : "default")'] ]
+            - [ setBar, ['@=service(''foo'').foo() ~ (container.hasParameter(''foo'') ? parameter(''foo'') : ''default'')'] ]
```
Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services6.yml]: This looks weird.

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services9.yml [hunk 3]
```diff
@@ -36,7 +36,7 @@ services:
             - [setBar, ['@?foo2']]
             - [setBar, ['@?foo3']]
             - [setBar, ['@?foobaz']]
-            - [setBar, ['@=service("foo").foo() ~ (container.hasparameter("foo") ? parameter("foo") : "default")']]
+            - [setBar, ['@=service(''foo'').foo() ~ (container.hasParameter(''foo'') ? parameter(''foo'') : ''default'')']]
```
Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services9.yml]: same here

File: src/Symfony/Component/DependencyInjection/Tests/Loader/XmlFileLoaderTest.php [hunk 4]
```diff
@@ -227,7 +227,7 @@ public function testLoadServices()
         $this->assertEquals('sc_configure', $services['configurator1']->getConfigurator(), '->load() parses the configurator tag');
         $this->assertEquals(array(new Reference('baz', ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE, false), 'configure'), $services['configurator2']->getConfigurator(), '->load() parses the configurator tag');
         $this->assertEquals(array('BazClass', 'configureStatic'), $services['configurator3']->getConfigurator(), '->load() parses the configurator tag');
-        $this->assertEquals(array(array('setBar', array()), array('setBar', array(new Expression('service("foo").foo() ~ (container.hasparameter("foo") ? parameter("foo") : "default")')))), $services['method_call1']->getMethodCalls(), '->load() parses the method_call tag');
+        $this->assertEquals(array(array('setBar', array()), array('setBar', array(new Expression("service('foo').foo() ~ (container.hasParameter('foo') ? parameter('foo') : 'default')")))), $services['method_call1']->getMethodCalls(), '->load() parses the method_call tag');
```
Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Tests/Loader/XmlFileLoaderTest.php]: What is the reason for this change?

File: src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php [hunk 5]
```diff
@@ -156,7 +156,7 @@ public function testLoadServices()
         $this->assertEquals('sc_configure', $services['configurator1']->getConfigurator(), '->load() parses the configurator tag');
         $this->assertEquals(array(new Reference('baz'), 'configure'), $services['configurator2']->getConfigurator(), '->load() parses the configurator tag');
         $this->assertEquals(array('BazClass', 'configureStatic'), $services['configurator3']->getConfigurator(), '->load() parses the configurator tag');
-        $this->assertEquals(array(array('setBar', array()), array('setBar', array()), array('setBar', array(new Expression('service("foo").foo() ~ (container.hasparameter("foo") ? parameter("foo") : "default")')))), $services['method_call1']->getMethodCalls(), '->load() parses the method_call tag');
+        $this->assertEquals(array(array('setBar', array()), array('setBar', array()), array('setBar', array(new Expression("service('foo').foo() ~ (container.hasParameter('foo') ? parameter('foo') : 'default')")))), $services['method_call1']->getMethodCalls(), '->load() parses the method_call tag');
```
Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php]: same here

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/containers/container9.php (hunk 1 above)
Comment (kalessil) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/containers/container9.php]: This due to code style fixes proposed by fabBot, it's lot of fun to repair UTs now...

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services6.yml (hunk 2 above)
Comment (kalessil) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services6.yml]: For sure.

**Comment by fabpot** [comment]
We should first make the changes on 2.6 as it is still maintained, instead of 2.7.

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/containers/container9.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/containers/container9.php]: Fabbot should not check fixtures files actually. It does not make sense. I suggest reverting this change

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services6.yml (hunk 2 above)
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/yaml/services6.yml]: please revert it. It does not make sense IMO

**Comment by stof** [comment]
@fabpot can you make fabbot ignore the DI fixtures containing generated files for comparison ? It does not make sense to check them for CS as they are generated files

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services9_compiled.php [hunk 6]
```diff
@@ -246,7 +246,7 @@ protected function getMethodCall1Service()
 
         $instance->setBar($this->get('foo'));
         $instance->setBar(NULL);
-        $instance->setBar(($this->get("foo")->foo() . (($this->hasparameter("foo")) ? ($this->getParameter("foo")) : ("default"))));
+        $instance->setBar(($this->get('foo')->foo().(($this->hasParameter('foo')) ? ($this->getParameter('foo')) : ('default'))));
```
Comment (kalessil) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services9_compiled.php]: @stof : this is code style changes, triggering all those fixtures adjustments - PR simply marked as not fulfilling code style requirements.

The same in Symfony/Component/DependencyInjection/Tests/Fixtures/php/services9.php

Comment (stof) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services9_compiled.php]: As said in my other comment, this is a bug in fabbot. It should not check fixtures files. Please just ignore the failures coming from fixtures files

**Comment by kalessil** [comment]
@fabpot : will do

File: src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services9_compiled.php (hunk 6 above)
Comment (kalessil) [inline src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services9_compiled.php]: I see, will revert.
